tests/tensor_search/test_add_documents.py

```diff
@@ -1450,4 +1451,45 @@ def test_determine_document_field_type(self):
             mappings=mixed_mappings
         ) == enums.DocumentFieldType.standard
             
-    
+    def test_determine_text_chunk_prefix(self):
+        """
+        Ensures proper priority order is followed when determining the chunk prefix.
+        add docs request-level > index override-level > model default level
+        """
+
+        index_info_with_model_default = IndexInfo(
+            model_name="prefix-test-model",
+            search_model_name=None,
+            properties={},
+            index_settings={
+                IndexSettingsField.index_defaults: {
+                    IndexSettingsField.model: "prefix-test-model",
+                    IndexSettingsField.treat_urls_and_pointers_as_images: True,
+                    IndexSettingsField.text_preprocessing: {}
+                }
+            }
+        )
+
+        index_info_with_override = IndexInfo(
+            model_name="prefix-test-model",
+            search_model_name=None,
+            properties={},
+            index_settings={
+                IndexSettingsField.index_defaults: {
+                    IndexSettingsField.model: "prefix-test-model",
+                    IndexSettingsField.treat_urls_and_pointers_as_images: True,
+                    IndexSettingsField.text_preprocessing: {
+                        IndexSettingsField.override_text_chunk_prefix: "index-override"
+                    }
+                }
+            }
+        )
+
+        # All prefixes on (request level chosen)
+        assert add_docs.determine_text_chunk_prefix("request-level", index_info_with_override) == "request-level"
```

Comment:
These are separate test cases. I suggest separate tests, each with a short docstring description (docsting for test cases comes up in test results).

Reply:
Used subTest to fix this problem